### PR TITLE
DSD-521: optionally display the helper/invalid text

### PR DESCRIPTION
**Pull request by EdwinGuzman**
Fixes JIRA ticket [DSD-521](https://jira.nypl.org/browse/DSD-521)

## This PR does the following:
- Adds a `showHelperInvalidText` flag to form-related component to optionally display the helper/invalid text in the `HelperErrorText` component.

### Checklist:

- [ ] I have updated the Storybook documentation accordingly.
- [ ] All new and existing tests passed.

### Front End Review:

- [ ] View [the example in Storybook]()

### CODE REVIEW

File: src/components/SearchBar/SearchBar.tsx [hunk 1]
```diff
@@ -83,7 +83,14 @@ export default function SearchBar(props: SearchBarProps) {
     textInputProps,
   } = props;
   const styles = useMultiStyleConfig("SearchBar", { hasSelect: selectProps });
-  const stateProps = { isDisabled, isInvalid, isRequired };
+  const stateProps = {
+    helperText: "",
+    isDisabled,
+    isInvalid,
+    isRequired,
+    showHelperInvalidText: false,
```
Comment (bigfishdesign13) [inline src/components/SearchBar/SearchBar.tsx:91]: I don't see this prop in the `SearchBarProps` interface.  How is this being used?

**Review by bigfishdesign13 — COMMENTED** [review]
I only have a question about what is happening in the files for `SearchBar`.  Everything else looks good.

File: src/components/SearchBar/SearchBar.tsx (hunk 1 above)
Comment (EdwinGuzman) [reply src/components/SearchBar/SearchBar.tsx:91]: `SearchBar` does not accept or use `showHelperInvalidText` as a prop. This is set to `false` in the `stateProps` object that is passed to both the `Select` and `TextInput` components. Those two components _do_ accept this prop. We just never want to show the helper/error text for `Select` or `TextInput` so they are always set to false.

Comment (bigfishdesign13) [reply src/components/SearchBar/SearchBar.tsx:91]: Got it.  Thanks.

**Review by bigfishdesign13 — APPROVED** [review]
Question answered.  Everything looks good.

**Pull request merged**